src/ctia/entity/vulnerability.clj

```diff
@@ -109,29 +111,93 @@
 (def vulnerability-enumerable-fields
   [:source])
 
+(defn build-query
+  [{:keys [part vendor product]}]
+  {:simple_query_string {:query (format "cpe\\:2.3\\:%s\\:%s\\:%s*"
+                                        part
+                                        (cpe/->simple-query-string-format vendor)
+                                        (cpe/->simple-query-string-format product))
+                         :flags "ESCAPE|PREFIX"
+                         :fields [:configurations.nodes.cpe_match.cpe23Uri
+                                  :configurations.nodes.children.cpe_match.cpe23Uri]}})
+
+(s/defn search-by-cpe-match-strings [{{:keys [get-store]} :StoreService
+                                      :as services} :- APIHandlerServices]
+  (context "/cpe_match_strings" []
+    (let [capabilities :search-vulnerability]
+      (GET "/" []
+        :return PartialVulnerabilityList
+        :query [{cpe23-match-strings :cpe23_match_strings limit :limit :as params}
+                (st/merge
+                 {:cpe23_match_strings (swagger-schema/describe [(s/constrained s/Str
+                                                                                (comp cpe/required-components-specified?
+                                                                                      cpe/->cpe-match
+                                                                                      ws/formatted-cpe-23-string?)
+                                                                                "Valid CPE Match with Product and Vendor components")]
+                                                                "CPE 2.3 match strings to search by")
+                  (s/optional-key :search_after) (swagger-schema/describe [s/Str] "Pagination stateless cursor")
+                  (s/optional-key :limit) (swagger-schema/describe Long "Pagination Limit")}
+                 BaseEntityFilterParams
+                 SourcableEntityFilterParams
+                 VulnerabilityFieldsParam)]
+        :summary "List Vulnerabilities with configurations matching CPE 2.3 match strings"
+        :description (routes.common/capabilities->description capabilities)
+        :capabilities capabilities
+        :auth-identity identity
+        :identity-map identity-map
+        (let [{{:keys [conn index]} :state} (get-store :vulnerability)
```

Comment:
the body of this route contains a several detail that should be tested. I would extract it and test it.